Explorer/Assets/DCL/Interaction/Raycast/Systems/ExecuteRaycastSystem.cs

```diff
@@ -90,13 +93,24 @@ private void BudgetAndExecute(Vector3 scenePosition)
         {
             // Process only not executed raycasts which bucket is not farther than the max allowed distance
             orderedData.Clear();
+            specialEntitiesData.Clear();
```

Comment:
I don't think you need a second collection as you can introduce a custom sorter that will serve `null` partition first